src/Transport/Addressing/Partitioning/Strategies/SingleNamespacePartitioning.cs

```diff
@@ -3,21 +3,17 @@ namespace NServiceBus
     using System.Collections.Generic;
     using System.Configuration;
     using System.Linq;
+    using AzureServiceBus.Addressing.Partitioning;
     using Settings;
     using Transport.AzureServiceBus;
 
     public class SingleNamespacePartitioning : INamespacePartitioningStrategy
     {
-        NamespaceConfigurations namespaces;
+        List<RuntimeNamespaceInfo> namespaces;
 
         public SingleNamespacePartitioning(ReadOnlySettings settings)
         {
-            if (!settings.TryGet(WellKnownConfigurationKeys.Topology.Addressing.Namespaces, out namespaces))
-            {
-                throw new ConfigurationErrorsException($"The '{nameof(SingleNamespacePartitioning)}' strategy requires exactly one namespace, please configure the connection string to your azure servicebus namespace.");
```

Comment:
We're losing the error message which would explain why strategy is not configured properly.